### PR TITLE
Add end to end Azure Functions annotation samples

### PR DESCRIPTION
**Pull request by JanusWu**
Tracking issue #130, #142 , #143.

### CODE REVIEW

**Comment by JanusWu** [comment]
@pragnagopa - could you review?  :)

**Comment by JanusWu** [comment]
I'm not sure why this PR contains previous commits, maybe I need to re-fork this repo?

**Comment by pragnagopa** [comment]
Can you rebase and squash your commits?

File: End2End/functest/src/main/java/com/microsoft/azure/functions/endtoend/ServiceBusTopic.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package com.microsoft.azure.functions.endtoend;
+
+import com.microsoft.azure.functions.ExecutionContext;
+import com.microsoft.azure.functions.annotation.BindingName;
+import com.microsoft.azure.functions.OutputBinding;
+
+public class ServiceBusTopic {
+
+    public static void serviceBusTopicTrigger(
+        @BindingName("message") String message,
```
Comment (pragnagopa) [inline End2End/functest/src/main/java/com/microsoft/azure/functions/endtoend/ServiceBusTopic.java]: Changes looks good. We do need to update all the tests to use annotations and delete checked in function.json files.

**Comment by JanusWu** [comment]
I'm confused. Delete all function.json files and only use annotations, you mean make the test like which in [Azure-Functions-Java-Samples](https://github.com/jdneo/Azure-Functions-Java-Samples/tree/master/ServiceBusQueueTrigger/project)?
But at the very beginning, we decided to move the annotation tests to azure-maven-plugins, and keep the tests with functions left. Right? May I know why we have this change?

Besides, If we do this change, the tests of HTTP, Queue, Timer and Blob which were already merged into java-worker also need to be changed. Right?
Correct me if I'm wrong.

**Comment by JanusWu** [comment]
@pragnagopa How about now?

**Comment by pragnagopa** [comment]
Commits look good :) Thanks! Can you confirm if the all tests in this PR are passing?

**Comment by JanusWu** [comment]
Add Cosmos DB test and squash the commit.

**Comment by JanusWu** [comment]
I have confirmed that all the tests are passing. :)

**Comment by pragnagopa** [comment]
Thanks @JanusWu. I will go ahead and merge them in. Will try out E2E tests as well.
cc: @paulbatum, @selvasingh  - We now have working versions of tests for all the triggers including CosmosDB.

**Pull request merged**